### PR TITLE
docs: Describe RPM package support in KubeRocketCI (#341)

### PR DESCRIPTION
**Pull request by Oleksandr123234**
## Description

This change add a new page to KubeRocketCI documentation that describe RPM package support implementation.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement
- [x] Documentation update

## How Has This Been Tested?

npm ci && npm start

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Pull Request contain one commit. I squash my commits.

### CODE REVIEW

File: docs/operator-guide/cd/deploy-rpm.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+The first approach to use RPM package is Default. This approach is considered more simple as it implies creating most of the resources via AWX user interface.
+
+### AWX Integration Flow
+
+Default approach involves the following steps:
+
+```mermaid
+graph LR;
+    A(Install AWX) --> B(Onboard Ansible Git repisitory) --> C(Configure AWX)--> D(Create application) --> E(Select deploy pipeline)
```
Comment (daniil-nedostup) [inline docs/operator-guide/cd/deploy-rpm.md:167]: A(Onboard Ansible Git repisitory) --> B(Install AWX)

Would this be more correct?

File: docs/operator-guide/cd/deploy-rpm.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+### AWX Integration Procedure
+
+Start off by ensuring that External Secrets Operator is [installed](../secrets-management/install-external-secrets-operator.md) and [configured](../secrets-management/external-secrets-operator-integration.md) properly.
+
+To set up RPM integration using AWX tool, follow the steps below. By completing these steps, you will be able to add, store, and deploy RPM applications:
+
+1. Onboard Ansible Git repository:
+
+  1. Create a repository where you will store you Ansible configuration files. Below is a structure of a GitOps repository designed for deploying RPM packages:
```
Comment (daniil-nedostup) [inline docs/operator-guide/cd/deploy-rpm.md:186]: Lets change "Below is a structure of a GitOps repository..." to "Below is a structure of an Ansible Git repository..."

File: docs/operator-guide/cd/deploy-rpm.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+4. Create application with the **rpm-package** deployment option. Refer to the [Add Application](../../user-guide/add-application.md) page for more details:
+
+  ![Create RPM application](../../assets/operator-guide/rpm/rpm.gif "Create RPM application")
+
+5. When [creating environments](../../user-guide/manage-environments.md#add-a-new-environment), in the **Deploy pipeline template** field, select **deploy-ansible-awx**:
+
+  ![Create Ansible environment](../../assets/operator-guide/rpm/ansible-deploy-ansible-awx.png "Create Ansible environment")
+
+6. When deploying application, enable the **Values override** option to apply configuration from the Git repository:
```
Comment (daniil-nedostup) [inline docs/operator-guide/cd/deploy-rpm.md:301]: 6. (Optional) When deploying a component, you can enable the **Values override** option if you want an additional role, created in the Ansible Git repository, to be executed specifically for this component. This role should follow the naming convention **\<codebase-name\>-\<deployment-flow-name\>-\<environment-name\>**.

File: versioned_docs/version-3.10/operator-guide/cd/deploy-rpm.md [hunk 4]
```diff
@@ -0,0 +1,309 @@
+# Deploy RPM Packages
+
+KubeRocketCI can use two types of deployment packages: Helm chart and RPM packages. While Helm chart is a default deployment package in KubeRocketCI, we recommend using RPM packages for specific Linux distributions, such as Oracle, Fedora, openSUSE, and other where these packages are supported and widespread. To learn more RPM packages, please refer to the official [documentation](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/rpm_packaging_guide/getting-started-with-rpm-packaging#introduction-to-rpm-packaging_getting-started-with-rpm-packaging).
+
+In KubeRocketCI, NPM support allows to collect applications, store them as Nexus artifacts, and deploy them using [Ansible](https://docs.ansible.com/ansible/latest/index.html) tool.
```
Comment (MykolaMarusenko) [inline versioned_docs/version-3.10/operator-guide/cd/deploy-rpm.md:5]: please fix  NPM, this guide about RPM

File: docs/operator-guide/cd/deploy-rpm.md [hunk 5]
```diff
@@ -0,0 +1,310 @@
+# Deploy RPM Packages
+
+KubeRocketCI can use two types of deployment packages: Helm chart and RPM packages. While Helm chart is a default deployment package in KubeRocketCI, we recommend using RPM packages for specific Linux distributions, such as Oracle, Fedora, openSUSE, and other where these packages are supported and widespread. To learn more RPM packages, please refer to the official [documentation](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/rpm_packaging_guide/getting-started-with-rpm-packaging#introduction-to-rpm-packaging_getting-started-with-rpm-packaging).
+
+In KubeRocketCI, NPM support allows to collect applications, store them as Nexus artifacts, and deploy them using [Ansible](https://docs.ansible.com/ansible/latest/index.html) tool.
+RPM package support implamentation is implemented in two approaches:
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:6]: please fix typo and duplication

File: docs/operator-guide/cd/deploy-rpm.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cd-ansible-gitops-key
+  namespace: <edp>
+data:
+  id_rsa: $(cat /path/to/repo_id_rsa | base64 | tr -d '\n')
+  url: $(echo -n "git@github.com:alezander86/ansible-gitops.git" | base64 | tr -d '\n')
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:58]: alezander86? we use this account?

File: docs/operator-guide/cd/deploy-rpm.md [hunk 7]
```diff
@@ -0,0 +1,310 @@
+# Deploy RPM Packages
+
+KubeRocketCI can use two types of deployment packages: Helm chart and RPM packages. While Helm chart is a default deployment package in KubeRocketCI, we recommend using RPM packages for specific Linux distributions, such as Oracle, Fedora, openSUSE, and other where these packages are supported and widespread. To learn more about RPM packages, please refer to the official [documentation](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/rpm_packaging_guide/getting-started-with-rpm-packaging#introduction-to-rpm-packaging_getting-started-with-rpm-packaging).
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:3]: lets remove 'we recommend'

File: docs/operator-guide/cd/deploy-rpm.md [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+```bash
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cd-ansible-gitops-key
+  namespace: <krci>
+data:
+  id_rsa: $(cat /path/to/repo_id_rsa | base64 | tr -d '\n')
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:57]: can i use these manifestos like this? is there a similar description in other guides?

Comment (Oleksandr123234) [reply docs/operator-guide/cd/deploy-rpm.md:56]: These manifests are valid, I've seen similar description in our documentation. For example:

* [Argo CD Integration](https://docs.kuberocketci.io/docs/operator-guide/cd/argocd-integration#keycloak-integration-optional)
* [Install via AWS Marketplace](https://docs.kuberocketci.io/docs/operator-guide/aws-marketplace-install#deploy-kuberocketci-platform)

File: docs/operator-guide/cd/deploy-rpm.md [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+
+In KubeRocketCI, RPM support allows you to collect applications, store them as Nexus artifacts, and deploy them using the [Ansible](https://docs.ansible.com/ansible/latest/index.html) tool. RPM package support is implemented using two approaches:
+
+**Default approach**: This approach involves configuring deployments using a GitOps repository and Kubernetes secrets. It is considered more complicated but flexible.
+**AWX approach**: This approach involves deploying applications using the [AWX](https://ansible.readthedocs.io/projects/awx/en/latest/) tool.
+
+## Default Approach
+
+The first approach to using RPM packages is the default approach. This approach is considered more complicated but efficient.
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:12]: duplication with line 7

File: docs/operator-guide/cd/deploy-rpm.md [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+Below is a breakdown of the GitOps repository structure:
+
+- **nexus.repo**: Сontains the configuration for connecting to Nexus as a repository on the instances.
+- **inventory.ini**: Contains groups with instances, user credentials, and additional settings for connecting to the instances.
+- **pre-deploy.yml**: Contains general playbooks that contain a set of tasks that must be done before starting work, such as copying and connecting Nexus config, installing additional packages depending on the group.
+
+The rest of the files are example of deployment configurations. To spread these configuration files between Deployment Flows and Environments, files comply to the following hierarchy:
+
+```bash
+\<deployment-flow-name\>/\<environment-name\>/\<package-name\>/01_\<file\>.yml
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:151]: can we use 'playbook' instead of <file>?

File: docs/operator-guide/cd/deploy-rpm.md [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+
+1. **Onboard Ansible Git repository**: Create an Ansible Git repository that contains playbooks that will be used in deploy pipelines.
+2. **Install AWX**: Set Up AWX application. We recommend using [cluster add-ons](../add-ons-overview.md) to install it.
+3. **Configure AWX**: Once AWX tool is installed, open its UI to create AWX Project, Inventory, Templates, etc.
+4. **Create application**: [Add your application](../../user-guide/add-application.md) to KubeRocketCI as an RPM package.
+5. **Select deploy pipeline**: Choose a specific deploy pipeline when creating/editing an environment to deploy RPM packages.
+
+### AWX Integration Procedure
+
+Start off by ensuring that External Secrets Operator is [installed](../secrets-management/install-external-secrets-operator.md) and [configured](../secrets-management/external-secrets-operator-integration.md) properly.
```
Comment (MykolaMarusenko) [inline docs/operator-guide/cd/deploy-rpm.md:179]: Prerequisite? Maybe we need to add this section?

**Review by MykolaMarusenko — APPROVED** [review]

**Review by MykolaMarusenko — APPROVED** [review]

**Pull request merged**